### PR TITLE
Update Angular from 13.0.0 to 14.0.0

### PR DESCRIPTION
**Pull request by mhenkens**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [X] The commit message follows our guidelines: https://github.com/NationalBankBelgium/stark/blob/master/CONTRIBUTING.md#-commit-message-guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[X] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Update angular version from 13.0.0 to angular 14.0.0

Issue Number: N/A

## Does this PR introduce a breaking change?

```
[X] Yes
[ ] No
```

See breaking change from ´Angular 14.0.0´ and ´TypeScript 4.6´

### CODE REVIEW

File: packages/stark-ui/src/util/form/form.util.ts [hunk 1]
```diff
@@ -62,9 +62,9 @@ export class StarkFormUtil {
 	 * @param formGroup - Angular form group object
 	 * @param statesToBeSet - States to be set to the form
 	 */
-	public static setFormGroupState(formGroup: FormGroup, statesToBeSet: StarkFormState[]): void {
+	public static setUntypedFormGroupState(formGroup: UntypedFormGroup, statesToBeSet: StarkFormState[]): void {
 		// Verifying it is indeed an Angular FormGroup
-		if (StarkFormUtil.isFormGroup(formGroup)) {
+		if (StarkFormUtil.isUntypedFormGroup(formGroup)) {
```
Comment (SuperITMan) [inline packages/stark-ui/src/util/form/form.util.ts:67]: Could you rename this back to `setFormGroupState` ?

File: packages/stark-ui/src/util/form/form.util.ts [hunk 2]
```diff
@@ -97,7 +97,7 @@ export class StarkFormUtil {
 	 * (calling `setFormControlState` function).
 	 * @param formGroup - Angular form group object
 	 */
-	public static isFormGroupValid(formGroup: FormGroup): boolean {
+	public static isUntypedFormGroupValid(formGroup: UntypedFormGroup): boolean {
```
Comment (SuperITMan) [inline packages/stark-ui/src/util/form/form.util.ts:100]: Could you rename this back to `isFormGroupValid` ?

File: packages/stark-ui/src/util/form/form.util.ts [hunk 3]
```diff
@@ -112,7 +112,7 @@ export class StarkFormUtil {
 	 * is an actual instance of an Angular {@link https://v12.angular.io/api/forms/FormGroup|FormGroup}.
 	 * @param formGroup - Angular form object
 	 */
-	public static isFormGroup(formGroup: any): formGroup is FormGroup {
+	public static isUntypedFormGroup(formGroup: any): formGroup is UntypedFormGroup {
```
Comment (SuperITMan) [inline packages/stark-ui/src/util/form/form.util.ts:115]: Could you rename this back to `isFormGroup` ?

**Review by SuperITMan — CHANGES_REQUESTED** [review]

File: packages/stark-ui/src/util/form/form.util.ts (hunk 1 above)
Comment (mhenkens) [reply packages/stark-ui/src/util/form/form.util.ts]: done

File: packages/stark-ui/src/util/form/form.util.ts (hunk 2 above)
Comment (mhenkens) [reply packages/stark-ui/src/util/form/form.util.ts]: done

**Review by SuperITMan — APPROVED** [review]

**Pull request merged**